### PR TITLE
ORLP-229 fix ExceptionHandlerController

### PR DESCRIPTION
**Pull request by Tamara20**

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/controller/ExceptionHandlerController.java [hunk 1]
```diff
@@ -33,16 +29,9 @@
...
-        return new ResponseEntity <>(new MessageDTO("The value of request parameter is not valid!"), HttpStatus.CONFLICT);
-    }
-
     @ExceptionHandler(RatingsBadValueException.class)
-    @ResponseBody
-    @ResponseStatus(value = HttpStatus.CONFLICT)
-    String handleRatingsBadValueException(HttpServletRequest request, Throwable ex) {
-        return "Rating can't be less than 1 and more than 5";
+    ResponseEntity <MessageDTO> handleRatingsBadValueException() {
+        return new ResponseEntity <>(new MessageDTO("Rating can't be less than 1 and more than 5"), HttpStatus.CONFLICT);
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/controller/ExceptionHandlerController.java]: How about IllegalArgumentException?

File: src/test/java/com/softserve/academy/spaced/repetition/controller/CardControllerTest.java [hunk 2]
```diff
@@ -68,14 +68,14 @@ public void getCardById() throws Exception {
...
 
-    @Test
-    public void getCardByWrongId() throws Exception {
-        when(cardService.getCard(eq(0L))).thenReturn(createCard());
-        mockMvc.perform(get("/api/card/{cardId}", 1L)
-                .accept(MediaType.APPLICATION_JSON)
-                .contentType(MediaType.APPLICATION_JSON))
-                .andExpect(status().isConflict());
-    }
+//    @Test
```
Comment (antonbakalets) [inline src/test/java/com/softserve/academy/spaced/repetition/controller/CardControllerTest.java]: Sonar: Commented code should be removed. If you ever need it, you will find it in Version Control System (Git).

Comment (Tamara20) [reply src/test/java/com/softserve/academy/spaced/repetition/controller/CardControllerTest.java]: ok I'll fix.

File: src/main/java/com/softserve/academy/spaced/repetition/controller/CardRatingController.java [hunk 3]
```diff
@@ -48,20 +47,20 @@
...
-     * @throws RatingsBadValueException
+     * @throws IllegalArgumentException
      * @throws NotAuthorisedUserException
      */
     @Auditable(action = AuditingAction.RATE_CARD)
     @PostMapping("/api/private/decks/{deckId}/cards/{cardId}/rate")
     @PreAuthorize(value = "@accessToUrlService.hasAccessToCard(#deckId, #cardId)")
-    public ResponseEntity<CardRatingPublicDTO> addCardRating(@RequestBody CardRating cardRating, @PathVariable Long deckId, @PathVariable Long cardId) throws RatingsBadValueException, NotAuthorisedUserException {
+    public ResponseEntity<CardRatingPublicDTO> addCardRating(@RequestBody CardRating cardRating, @PathVariable Long deckId, @PathVariable Long cardId) throws NotAuthorisedUserException {
         if ((cardRating.getRating() >= MIN_RATING) && (cardRating.getRating() <= MAX_RATING)) {
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/controller/CardRatingController.java]: This check should be on service level.

File: src/main/java/com/softserve/academy/spaced/repetition/controller/CourseRatingController.java [hunk 4]
```diff
@@ -35,12 +34,12 @@
...
 
     @PostMapping("/api/private/course/{courseId}")
-    public ResponseEntity addCourseRating(@RequestBody RatingDTO ratingDTO, @PathVariable Long courseId) throws RatingsBadValueException, NotAuthorisedUserException, UserStatusException {
+    public ResponseEntity addCourseRating(@RequestBody RatingDTO ratingDTO, @PathVariable Long courseId) throws NotAuthorisedUserException, UserStatusException {
         if ((ratingDTO.getRating() >= MIN_RATING) && (ratingDTO.getRating() <= MAX_RATING)) {
             courseRatingService.addCourseRating(ratingDTO.getRating(), courseId);
            return new ResponseEntity(HttpStatus.CREATED);
         } else {
-            throw new RatingsBadValueException();
+            throw new IllegalArgumentException("Rating can't be less than 1 and more than 5");
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/controller/CourseRatingController.java]: And it appears that hte same check is used in three different places. Lets create a separate issue.

**Pull request merged**